docs/guidance/rayservice.md

```diff
@@ -0,0 +1,158 @@
+
+For the fruit example deployment, you can try the following request
+```shell
+[ root@curl:/ ]$ curl  -X POST -H 'Content-Type: application/json' rayservice-sample-serve-svc.default.svc.cluster.local:8000 -d '["MANGO", 2]'
+6
+```
+You can get the response as `6`.
+
+* Note: serve-svc will do traffic routing among all the workers which have serve deployments. It is HA in general.
+* Note: serve-svc will always try it best to point to the healthy cluster, even during upgrading or failing cases.
+* Note: You can set `serviceUnhealthySecondThreshold` to define the threshold of seconds that the serve deployments fail.
+* Note: You can set `deploymentUnhealthySecondThreshold` to define the threshold of seconds that the Ray fails to deploy any serve deployments.
+
+### Update Ray Serve Deployment Graph
+
+You can update the `serveDeploymentGraphConfig` in your RayService config file.
+For example, if you update the mango price to 4 in [ray_v1alpha1_rayservice.yaml](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/ray_v1alpha1_rayservice.yaml).
+```shell
+  - name: MangoStand
+    numReplicas: 1
+    userConfig: |
+      price: 4
+```
+
+Do a `kubectl apply` to update your RayService.
+
+You can check the kubernetes stats of your RayService. It should show similar:
+```shell
+    serveDeploymentStatuses:
+    - healthLastUpdateTime: "2022-07-18T21:51:37Z"
+      lastUpdateTime: "2022-07-18T21:51:41Z"
+      name: MangoStand
+      status: UPDATING
+```
+
+After it finishes deployment, let's send a request again.
+```shell
+[ root@curl:/ ]$ curl  -X POST -H 'Content-Type: application/json' rayservice-sample-serve-svc.default.svc.cluster.local:8000 -d '["MANGO", 2]'
+8
+```
+Now you will get `8` as a result.
+
+### Upgrade RayService RayCluster Config
+You can update the `rayClusterConfig` in your RayService config file.
+For example, you can increase the worker node num to 2.
+```shell
+workerGroupSpecs:
+  # the pod replicas in this group typed worker
+  - replicas: 2
+```
+
+Do a `kubectl apply` to update your RayService.
+
+You can check the kubernetes stats of your RayService. It should show similar:
+```shell
+  pendingServiceStatus:
+    appStatus: {}
+    dashboardStatus:
+      healthLastUpdateTime: "2022-07-18T21:54:53Z"
+      lastUpdateTime: "2022-07-18T21:54:54Z"
+    rayClusterName: rayservice-sample-raycluster-bshfr
+    rayClusterStatus: {}
+```
+You can see RayService is preparing a pending cluster. After the pending cluster is healthy, RayService will switch it as active cluster and terminate the previous cluster.
+
+### RayService Observability
+RayService is native kubernetes CRD. You can use `kubectl logs` to check the operator logs or the head/worker nodes logs.
+You can also use `kubectl describe rayservices rayservice-sample` to check the states and event logs of your RayService instance.
+
+You can also login the head pod and use Ray cli to check the logs.
```

Comment:
Could you elaborate on this with the commands that would produce the logs?

Reply:
@iycheng Do you have a link to Ray observability?

Reply:
We have a doc internally. But the public url will be added by the end of this week. After that, I can have a followup pr for it.